### PR TITLE
Support OpenSSL3 with "no-deprecated" build option

### PR DESCRIPTION
**Pull request by trengginas**
When building OpenSSL with "no-deprecated" option, the library will fail due to use of deprecated method.
- `SSL_get_peer_certificate()`
- `X509_get_notBefore()`
- `X509_get_notAfter()`

This PR will fix the error by using the recommended/updated method.

This PR will also update the bundled `libsrtp` since it might also use OpenSSL.

Due to the changes, some updates also needed to support AES-GCM in SRTP (#1943).
The updates are required in Visual Studio (no changes on GNU targets).

1. [Install OpenSSL](https://docs.pjsip.org/en/latest/get-started/windows/build_instructions.html#installing-openssl).

1. In Visual Studio, go to `libsrtp` project, into folder crypto.

1. Multiple select `cipher/aes_gcm_ossl.c, cipher/aes_icm_ossl.c` 

   Right-click, then click Properties. Change Configuration to "All Configurations" and Platform to "All Platforms". Go to 
   Configuration Properties - General, then change Excluded From Build from "Yes" to "No".

1. Select `cipher/aes_icm.c`

   Right-click, then click Properties. Change Configuration to "All Configurations" and Platform to "All Platforms". Go to 
   Configuration Properties - General, then change Excluded From Build from "No" to "Yes".

1. Click on `libsrtp` project.

   Right-click, then click Properties. Change Configuration to "All Configurations" and Platform to "All Platforms". Go to 
   Configuration Properties - C/C++ - Preprocessor, then choose Preprocessor Definitions, and add `";OPENSSL;GCM"`.

1. Enable the desired AES GCM cryptos in your config_site.h, for example:
    ```
    #define PJMEDIA_SRTP_HAS_AES_GCM_256      1
    #define PJMEDIA_SRTP_HAS_AES_GCM_128      1
    ```

1. Rebuild `libsrtp`.

### CODE REVIEW

File: third_party/srtp/PJSIP_NOTES [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Local changes:
+1. Error build "Platform not recognized"
```
Comment (sauwming) [inline third_party/srtp/PJSIP_NOTES:2]: need to be more specific here. On which platform(s) did it fail?

File: pjlib/src/pj/ssl_sock_ossl.c [hunk 2]
```diff
@@ -100,6 +100,17 @@
 
 #   include <openssl/obj_mac.h>
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L || \
+    (defined(OPENSSL_API_COMPAT) && OPENSSL_API_COMPAT >= 0x10100000L)
+#  define X509_get_notBefore(x)     X509_getm_notBefore(x)
```
Comment (sauwming) [inline pjlib/src/pj/ssl_sock_ossl.c:105]: There are similar definitions below, so perhaps combine them, to make sure there's no redefinition warning/error?

**Comment by sauwming** [comment]
> Due to the changes, some updates also needed to support AES-GCM in SRTP (#1943). The updates are required in Visual Studio (no changes on GNU targets).
> 
> 1. [Install OpenSSL](https://docs.pjsip.org/en/latest/get-started/windows/build_instructions.html#installing-openssl).
> 2. In Visual Studio, go to `libsrtp` project, into folder crypto.
> 3. Multiple select `cipher/aes_gcm_ossl.c, cipher/aes_icm_ossl.c`
>    Right-click, then click Properties. Change Configuration to "All Configurations" and Platform to "All Platforms". Go to
>    Configuration Properties - General, then change Excluded From Build from "Yes" to "No".
> 4 ....

Are all these steps supposed to be done by user??

File: third_party/srtp/PJSIP_NOTES (hunk 1 above)
Comment (trengginas) [reply third_party/srtp/PJSIP_NOTES:2]: The reason is build error on every platform.
On `datatypes.h` there's a check for certain config. 
(https://github.com/pjsip/pjproject/blob/affc21f07cf610a861b3da82b96e41542b9bb4b4/third_party/srtp/crypto/include/datatypes.h#L57-L63).

The current code doesn't check if those configs are not defined.
https://github.com/pjsip/pjproject/blob/a5023f7fdd4506808e3cc50da2fdbf7b47bdda0b/third_party/srtp/crypto/include/datatypes.h#L57-L63

These configs are defined in `srtp_config.h`. 
Alternatively we can add `#include <srtp_config.h>` to `transport_srtp.c`
https://github.com/pjsip/pjproject/blob/affc21f07cf610a861b3da82b96e41542b9bb4b4/pjmedia/src/pjmedia/transport_srtp.c#L101

This way no changes to datatypes.h is needed

**Review by sauwming — APPROVED** [review]

**Review by nanangizz — APPROVED** [review]

**Pull request merged**